modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/DateFormat.java

```diff
@@ -115,6 +114,8 @@ Function<String, ZonedDateTime> getFunction(String format, ZoneId zoneId, Locale
         }
     };
 
+    private static final DateFormatter ISO_8601_DATE_FORMATTER = DateFormatter.forPattern("iso8601");
```

Comment:
Maybe a comment on this constant that justifies its existence? I'm trying to prevent some future engineer from thinking "oh, this constant is only just used the once, I'll simplify by removing it." and then undoing this change.